### PR TITLE
[c10d][doc] Add docs for ENV variables TORCH_NCCL_ASYNC_ERROR_HANDLING TORCH_NCCL_TRACE_CPP_STACK and TORCH_NCCL_COORD_CHECK_MILSEC

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132920

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: docs/source/torch_nccl_environment_variables.rst [hunk 1]
```diff
@@ -9,6 +9,8 @@ For more information on the environment variables, see `ProcessGroupNCCL Environ
 
   * - Variable
     - Description
+  * - ``TORCH_NCCL_ASYNC_ERROR_HANDLING``
+    - Control how whether or not we perform Async Error Handling with NCCL when an exception is observed in watchdog. If set to 0, no handling of asynchronous NCCL errors. If set to 1, aborting NCCL communicator and tearing down process upon error. If set to 2, only abort NCCL communicator and if set to 3, tearing down process without aborting NCCL communicator. By default, it is set to 3.
```
Comment (fegin) [inline docs/source/torch_nccl_environment_variables.rst:13]: There is an extra how? `Control whether or not`

File: docs/source/torch_nccl_environment_variables.rst [hunk 2]
```diff
@@ -25,6 +27,10 @@ For more information on the environment variables, see `ProcessGroupNCCL Environ
     - Control the watchdog heartbeat timeout period after which the monitoring thread will abort the process.
   * - ``TORCH_NCCL_TRACE_BUFFER_SIZE``
     - The maximum number of events we store in the flight recorder's ring buffer. One event could be the start or end of a collective, for example. Set to 0 to disable the tracebuffer and debugging info dump.
+  * - ``TORCH_NCCL_TRACE_CPP_STACK``
+    - Whether to collect cpp stack traces for flight recorder.
```
Comment (fegin) [inline docs/source/torch_nccl_environment_variables.rst:31]: Do we have a default value?

File: docs/source/torch_nccl_environment_variables.rst [hunk 3]
```diff
@@ -25,6 +27,10 @@ For more information on the environment variables, see `ProcessGroupNCCL Environ
...
+  * - ``TORCH_NCCL_COORD_CHECK_MILSEC``
+    - Control the interval inside the monitoring thread to check the coordinated signal from other ranks, e.g. to dump the debugging information.
```
Comment (fegin) [inline docs/source/torch_nccl_environment_variables.rst:33]: What's the default value?

**Review by fegin — APPROVED** [review]

**Review by fegin — APPROVED** [review]

**Review by wconstab — APPROVED** [review]
:)

Note for later: can we actually do a rollout to get everyone on async mode 3? And then delete this env?

File: docs/source/torch_nccl_environment_variables.rst (hunk 2 above)
Comment (fduwjj) [reply docs/source/torch_nccl_environment_variables.rst:31]: Sure, let me add it.

File: docs/source/torch_nccl_environment_variables.rst (hunk 3 above)
Comment (fduwjj) [reply docs/source/torch_nccl_environment_variables.rst:33]: Sure, let me add it.

**Comment by fduwjj** [comment]
@wconstab sure... let me add a TODO here first maybe?

**Pull request closed without merging**